### PR TITLE
Adding user-agent to GCS client in Python

### PR DESCRIPTION
**Pull request by pabloem**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by pabloem** [comment]
Run Python 3.8 PostCommit

**Comment by pabloem** [comment]
r: @chamikaramj

File: sdks/python/apache_beam/io/gcp/gcsio.py [hunk 1]
```diff
@@ -155,7 +156,10 @@ def __init__(self, storage_client=None):
           credentials=auth.get_service_credentials(),
           get_credentials=False,
           http=get_new_http(),
-          response_encoding='utf8')
+          response_encoding='utf8',
+          additional_http_headers={
+              "User-Agent": "apache-beam-%s" % apache_beam.__version__
+          })
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/gcsio.py:162]: Will it be possible to add a unit test to make sure this stays consistent and does not get broken ?

**Review by chamikaramj — COMMENTED** [review]
Thanks Pablo.

File: sdks/python/apache_beam/io/gcp/gcsio.py [hunk 2]
```diff
@@ -155,7 +156,10 @@ def __init__(self, storage_client=None):
           credentials=auth.get_service_credentials(),
           get_credentials=False,
           http=get_new_http(),
-          response_encoding='utf8')
+          response_encoding='utf8',
```
[the remark this answers is not in the document]
Comment (pabloem) [reply sdks/python/apache_beam/io/gcp/gcsio.py:159]: added for BQ

File: sdks/python/apache_beam/io/gcp/gcsio.py (hunk 1 above)
Comment (pabloem) [reply sdks/python/apache_beam/io/gcp/gcsio.py:162]: added tests for BQ and GCS

**Comment by pabloem** [comment]
Run Python 3.8 PostCommit

**Comment by pabloem** [comment]
Postcommit: https://ci-beam.apache.org/job/beam_PostCommit_Python38_PR/449/

**Comment by pabloem** [comment]
PTAL : D

File: sdks/python/apache_beam/io/gcp/bigquery_tools_test.py [hunk 3]
```diff
@@ -220,6 +220,18 @@ def test_delete_dataset_retries_for_timeouts(self, patched_time_sleep):
...
     self.assertTrue(client.datasets.Delete.called)
 
+  @mock.patch('time.sleep', return_value=None)
+  @mock.patch('google.cloud._http.JSONConnection.http')
+  def test_user_agent_insert_all(self, http_mock, patched_sleep):
+    wrapper = beam.io.gcp.bigquery_tools.BigQueryWrapper()
+    try:
+      wrapper._insert_all_rows('p', 'd', 't', [{'name': 'any'}], None)
+    except:  # pylint: disable=bare-except
+      # Ignore errors
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigquery_tools_test.py:230]: Let's explain why :)

File: sdks/python/apache_beam/io/gcp/bigquery_tools_test.py [hunk 4]
```diff
@@ -242,6 +254,19 @@ def test_temporary_dataset_is_unique(self, patched_time_sleep):
...
 
+  @mock.patch('time.sleep', return_value=None)
+  def test_user_agent_passed(self, sleep_mock):
+    wrapper = beam.io.gcp.bigquery_tools.BigQueryWrapper()
+    request_mock = mock.Mock()
+    wrapper.client._http.request = request_mock
+    try:
+      wrapper.create_temporary_dataset('project_id', 'location')
+    except:  # pylint: disable=bare-except
+      # Ignore errors
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigquery_tools_test.py:265]: Ditto.

File: sdks/python/apache_beam/io/gcp/gcsio_test.py [hunk 5]
```diff
@@ -420,6 +420,19 @@ def test_delete(self):
...
+  @mock.patch(
+      'apache_beam.io.gcp.gcsio.auth.get_service_credentials',
+      wraps=lambda: None)
+  @mock.patch('apache_beam.io.gcp.gcsio.get_new_http')
+  def test_user_agent_passed(self, get_new_http_mock, get_service_creds_mock):
+    client = gcsio.GcsIO()
+    try:
+      client.get_bucket('mabucket')
+    except:  # pylint: disable=bare-except
+      pass
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/gcsio_test.py:432]: Ditto.

**Review by chamikaramj — COMMENTED** [review]
Thanks Pablo. LGTM.

**Pull request merged**